### ISSUE TITLE
%in% method for rast object

### ISSUE THREAD

**Issue by shoebodh** [rspatial/terra#97]
In the raster package, there is a match() method for raster objects so the following works neatly when trying to generate a simple binary raster object from an input raster with several values 
Following works
library(raster)
```r
ras = raster(matrix(1:100, 10, 10))
#create 0, 1 raster directly using %in% function, pretty quick and easy
tras = ras %in% 1:5
raster::values(tras) = as.integer(raster::values(tras))
```
#but in terra the following fails
```r
library(terra)
ras2 = rast(matrix(1:100, 10, 10))
tras2 = ras2 %in% 1:5
Error in match(x, table, nomatch = 0L) : 
  'match' requires vector arguments
```
It would be pretty useful to have the %in% method for rast objects as well

**Comment by rhijmans** [comment]
Thank you, that was an oversight. You can now do 

```
library(terra)
ras2 = rast(matrix(1:100, 10, 10))
tras2 = ras2 %in% 1:5
tras2
#class       : SpatRaster 
#dimensions  : 10, 10, 1  (nrow, ncol, nlyr)
#resolution  : 1, 1  (x, y)
#extent      : 0, 10, 0, 10  (xmin, xmax, ymin, ymax)
#coord. ref. : +proj=longlat +datum=WGS84 +no_defs 
#data source : memory 
#names       : lyr.1 
#min values  :     0 
#max values  :     1 
```

**Comment by shoebodh** [comment]
Thanks for that quick fix. However, I noticed that for fairly large raster (my raster was around 8000x9000), the SpatRaster object is very slow (in fact it never completed this task) whereas the RasterLayer object does a decent job in terms of speed.
for example
```r
library(terra)
library(raster)

ras1 = raster(mat1 )
ras2 = ras1 %in% c(1:3) # This is a pretty quick operation
system.time(ras1 %in% c(1:3))

mat1 = matrix(as.integer(runif(n = 8000*9000, 1, 5)), 8000, 9000)
tras1 = rast(mat1)
tras2 = tras1 %in% c(1:3) #this takes too long, actually had to interupt after about a minute
```
Changing the rast() object raster will then work, like
```r
system.time(raster(tras1) %in% c(1:3))
   user  system elapsed 
   3.84    2.08    5.92 
```

**Comment by rhijmans** [comment]
Interesting. It is much better now. I get

```
n = 8000
mat1 = matrix(as.integer(runif(n = n*n, 1, 5)), n, n)

library(raster)
ras1 <- raster(mat1 )
system.time(ras2 <- ras1 %in% 1:3)
#   user  system elapsed 
#   0.78    0.05    0.83 
 
library(terra)
tras1 <- rast(mat1)
system.time(tras2 <- tras1 %in% 1:3)
#   user  system elapsed 
#   1.09    0.17    1.27 
``` 

`raster` is still faster, but not by much. It is an interesting exception to the rule that I should look into.

**Comment by shoebodh** [comment]
Really appreciate such quick responses. I agree, this is interesting; I was definitely expecting a faster execution for terra, based on the amazing gains in speed of crop(), aggregate and project() function (the ones I have used the most)
But  this %in% is manageable now. So, thanks.
Personally, I think, terra has an amazing potential but being in early stages, lacks the level of coverage raster provides. Lot of the issues I encountered are already on open issues lists, e.g., direct support for sf, and sp objects, like the raster provides. 
Hope to see these features in the coming versions. 
Thanks again.
